src/main/java/org/springframework/data/elasticsearch/core/MappingBuilder.java

```diff
@@ -333,6 +355,28 @@ private static void addFieldMappingParameters(XContentBuilder builder, Object an
 		}
 	}
 
+	/**
+	 * Apply mapping for dynamic templates.
+	 *
+	 * @throws IOException
+	 */
+	private static void addDynamicTemplatesMapping(XContentBuilder builder, String jsonString) throws IOException {
+		JsonNode jsonNode = new ObjectMapper().readTree(jsonString).get("dynamic_templates");
+		if (jsonNode != null && isNotBlank(jsonNode.toString()) && jsonNode.isArray()){
+			builder.field(FIELD_DYNAMIC_TEMPLATES);
+			builder.startArray();
+			Iterator<JsonNode> iterator = jsonNode.elements();
+			while (iterator.hasNext()){
+				String next = iterator.next().toString();
+				if (iterator.hasNext()){
+					next = next.concat(",");
+				}
+				builder.rawValue(new BytesArray(next.getBytes()), XContentType.JSON);
```

Comment:
Not sure if we really need to iterate over all node elements. It should be enough to write the `jsonNode` as raw value.

```java
String json = objectMapper.writeValueAsString(dynamicTemplate).getBytes();
builder.field(FIELD_DYNAMIC_TEMPLATES);
builder.rawValue(new BytesArray(jsonNode.getBytes()), XContentType.JSON);
```

Reply:
Nice simplification. Applied.